### PR TITLE
Configurable Logger Backend

### PR DESCRIPTION
**Pull request by mrbaboon**
Backends are now configurable in settings.

This enables the sending of logs directly to a configured python logger, avoiding database bloat when there are lots of modifications. The default settings should leave everything existing untouched (defaults to ModelBackend), but we can now configure a logger backend to direct auditlogs to a configured logger.

I haven't included a log formatter, as that is up to the end user to decide how they want to format them (we'll be doing JSON, to be consumed by our ELK stack). All of the audit metadata gets dumped into the logs extra for ease of extraction when formatting a log entry.

I've had to leverage a bit more data in thread storage as I didn't want to make another trip to the database to fetch the actor details when performing the audit logging. Felt a little bad but let me know what you think.

Could definitely use some Readme/docs updates.  I can look into doing that.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,9 +1,11 @@
 from distutils.core import setup
 
+from setuptools import find_packages
+
 setup(
     name='django-auditlog',
     version='0.4.5',
-    packages=['auditlog', 'auditlog.migrations', 'auditlog.management', 'auditlog.management.commands'],
+    packages=find_packages(where='src', exclude=['*tests*', 'runtests']),
```
Comment (audiolion) [inline setup.py]: I am ok switching to `find_packages` but I am concerned that the `exclude` is only excluding `src/auditlog_tests/tests.py` and not the entire `src/auditlog_tests`.

File: src/auditlog/backends/logger.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+from auditlog import settings
+from auditlog.models import LogEntry
+from auditlog.backends.base import AuditBackend
+from auditlog.middleware import get_thread_data
+
+
+class LoggerBackend(AuditBackend):
+    """Python logger based backend for creation of audit log entries."""
+    message = '{action} on {content_type_name} {object_pk} by actor {actor_name}'
+    logger = 'audit.log'
```
Comment (audiolion) [inline src/auditlog/backends/logger.py]: what does this do? I am not seeing references to `self.logger` and `audit.log` is set as the default in `settings.AUDITLOG_LOGGER`. Also, what is the `audit.log` default logger name related to? Any reason it needs to be a descendent of `audit`?

**Comment by mrbaboon** [comment]
Thanks @audiolion just saw this review.

I'll plug away at these when I get time at work over the next few days.

File: setup.py (hunk 1 above)
Comment (mrbaboon) [reply setup.py]: I can tweak it.

File: src/auditlog/backends/logger.py (hunk 2 above)
Comment (mrbaboon) [reply src/auditlog/backends/logger.py]: I think it was legacy - was using during development but then I moved to a package setting.

**Comment by mrbaboon** [comment]
PS. great review!

**Comment by dsever** [comment]
Hi, 

Is this in plan to me merged?

**Comment by mrbaboon** [comment]
@dsever we can close it or somebody else will have to pick it up.  Unfortunately I haven't been able to keep it up to date.

**Comment by dsever** [comment]
Now I'm very sad :( I would like to have this feature

**Comment by mrbaboon** [comment]
@dsever feel free to update the PR and take it over :D  We aren't using this package for audit logging anymore so I can't validate the time to get it back on track to merge.

**Comment by cansin** [comment]
Ah, this is exactly what I was researching for as well. i.e. bypassing DB calls in favor of directly hitting a logger. @hramezani do you know if this is in works somewhere?

**Pull request closed without merging**